src/utils/walletConnect.ts

```diff
@@ -0,0 +1,263 @@
+import SignClient from '@walletconnect/sign-client';
+import { SignClientTypes } from '@walletconnect/types';
+import { getSdkError, parseUri } from '@walletconnect/utils';
+import { WC_PROJECT_ID } from 'react-native-dotenv';
+import { NavigationContainerRef } from '@react-navigation/native';
+import Minimizer from 'react-native-minimizer';
+
+import { logger, RainbowError } from '@/logger';
+import { WalletconnectApprovalSheetRouteParams } from '@/redux/walletconnect';
+import { Navigation } from '@/navigation';
+import { getActiveRoute } from '@/navigation/Navigation';
+import Routes from '@/navigation/routesNames';
+import { analytics } from '@/analytics';
+import { maybeSignUri } from '@/handlers/imgix';
+import { dappLogoOverride, dappNameOverride } from '@/helpers/dappNameHandler';
+import { Alert } from '@/components/alerts';
+import * as lang from '@/languages';
+
+/**
+ * Indicates that the app should redirect or go back after the next action
+ * completes
+ *
+ * This is a hack to get around having to muddy the scopes of our event
+ * listeners. BE CAREFUL WITH THIS.
+ */
+let hasDeeplinkPendingRedirect = false;
+
+/**
+ * Set `hasDeeplinkPendingRedirect` to a boolean, indicating that the app
+ * should redirect or go back after the next action completes
+ *
+ * This is a hack to get around having to muddy the scopes of our event
+ * listeners. BE CAREFUL WITH THIS.
+ */
+export function setHasPendingDeeplinkPendingRedirect(value: boolean) {
+  logger.info(`setHasPendingDeeplinkPendingRedirect`, { value });
+  hasDeeplinkPendingRedirect = value;
+}
+
+const signClient = Promise.resolve(
+  SignClient.init({
+    projectId: WC_PROJECT_ID,
+    // relayUrl: "<YOUR RELAY URL>",
+    metadata: {
+      name: '🌈 Rainbow',
```

Comment:
idk about this emoji here but its also cute so 🤷

Reply:
Haha fair, this is just how we have it set up for v1 lol, I copied